### PR TITLE
Added Circular Saw Blade

### PR DESCRIPTION
**Pull request by jDenyes**

### CODE REVIEW

File: godotproject/SawBlade.gd
Comment (charasyn) [inline godotproject/SawBlade.gd]:
```suggestion

# In order to stop the tongue when it hits the sawblade, we have our "tongue_can_damage" collision bit set.
# However, the saw shouldn't take damage from the tongue, so we have this empty function.
func take_damage(damage):
	pass
```

File: godotproject/SawBlade.gd
Comment (charasyn) [inline godotproject/SawBlade.gd]:
```suggestion
```

File: godotproject/SawBlade.gd
Comment (charasyn) [inline godotproject/SawBlade.gd]:
```suggestion
```

File: godotproject/SawBladeArea.gd
Comment (charasyn) [inline godotproject/SawBladeArea.gd]: Can you remove this unused script?

File: godotproject/SawBladeArea.tscn
Comment (charasyn) [inline godotproject/SawBladeArea.tscn]: I don't see the timer used anywhere in the code.. can it be removed?

**Review by charasyn — CHANGES_REQUESTED** [review]
Hey, there are a few things that I think should be cleaned up before being checked in. In addition, I think the files should be moved into a folder. As per teams, I'll move it and change the image size

**Review by charasyn — COMMENTED** [review]
Updated with all the changes I asked for - no functional changes, just some cleanup and resizing of the sprite. Can someone double-check

**Review by charasyn — APPROVED** [review]

**Comment by charasyn** [comment]
Uh oh. I think there are two copies of the sawblade now. I think I probably shouldn't have force pushed...

**Comment by fheysel** [comment]
There are 2 tscn and gd files for the sawblade. Need to resolve and delete one of them before we can merge

**Comment by fheysel** [comment]
@jDenyes  or @charasyn could one of you resolve the duplication so we can merge this please?

**Comment by charasyn** [comment]
Okay I force pushed again without squashing the commits this time. The branch should be in a good state so it should be ready to merge.

**Comment by charasyn** [comment]
Wait this doesn't have Jacob's changes that he was working on to change how the sawblade moves. I'll message him to sort it out

**Comment by jDenyes** [comment]
adding audio file as Fiona suggested, waiting for conversion from mp3 to godot supported format.

**Review by charasyn — CHANGES_REQUESTED** [review]
After talking with Jacob, I'm making some modifications locally to fix some things

**Review by charasyn — APPROVED** [review]
OK I added my changes. @jDenyes can you please look over the commit that I made and approve it? If it's ok with you then it should be good to merge.

**Comment by jDenyes** [comment]
@charasyn thanks for the cleanup. I actually encountered a bug while testing rn so I'm gonna fix it quickly. framerate drops lead to weird behavior.

**Comment by charasyn** [comment]
Make sure you pull this and check to make sure it's still there. I actually have FPS drops when running the game in-editor so I moved the code from _process to _physics_process

**Comment by jDenyes** [comment]
What's the difference between the two?

**Comment by jDenyes** [comment]
It actually looks good now, but it's hard to reproduce. Sometimes the sawblade would just yeet to infinity, but looking at the logic it's hard for me to understand why.

**Comment by charasyn** [comment]
In _physics_process the delta is always the same, there's always 60 physics process per second. However if the physics code is put in _process, it can result in it being used with a very large delta if there's a lag spike. Since we rely on the velocity being computed and applied smoothly, it can cause a very large displacement if it's not done smoothly.

For example: suppose the sawblade was just about to reach the point where it starts decelerating. At this point it has max speed. Now suppose that there's a lag spike that results in the next delta being 1s instead of 16ms. This means that the displacement calculated is going to be 60x what it should have been, AND the sawblade hasn't even started decelerating yet. So it's gonna go way further than it should have and it still needs to decelerate after it moves that far.

Using _physics_process completely avoids this issue by just running the process multiple times to "catch up" after a lag spike (I think)
________________________________

**Comment by jDenyes** [comment]
Ok with those changes I'm happy to merge to main. @fheysel to set the spawn point edit the StartPos, to change where it goes to edit the EndPos. Come work with me to integrate. I'm putting it in now

**Pull request merged**
